tests/integration/ha/helpers_data.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper functions for data related tests, such as indexing, searching etc.."""
```

Comment:
i'm not the most familiar with opensearch calls and such @WRFitch might be able to better review this file